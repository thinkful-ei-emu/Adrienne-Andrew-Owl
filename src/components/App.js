import React from 'react';

import '../styles/App.css';
import '../styles/ParticipantList.css';

import ParticipantList from './ParticipantList';
import ChatEvents from './ChatEvents';
import Stage from './Stage';

function App(props) {
  return (
    <div className="App">
      <ParticipantList
        participants={props.participants}
      />
      <hr />
      <ChatEvents
        participants={props.participants}
        chatEvents={props.chatEvents}
      />
      <hr />
      <Stage 
        participants={props.participants}
      />
    </div>
  );
}

export default App;