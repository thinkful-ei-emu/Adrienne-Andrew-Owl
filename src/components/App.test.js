import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import participants from '../data/participantData';
import chatEvents from '../data/chatEventsData';

// smoke test; failing time
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App
    participants={[]}
    chatEvents={[]}
    />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// screenshot test;
it('renders this UI as expected', () => {
  const appTest = renderer.create(<App
    participants={participants}
    chatEvents={chatEvents}
    />).toJSON();
  // Check whether it matches the previous snapshot
  // Stored in __snapshots__/App.test.js.snap
  expect(appTest).toMatchSnapshot(); 
});