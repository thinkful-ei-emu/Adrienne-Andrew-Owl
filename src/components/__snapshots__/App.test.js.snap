// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `
<div
  className="App"
>
  <div
    className="participantList"
  >
    <div
      className="participant"
    >
      <img
        alt="Avatar Here but no Avatar, sucks"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <p>
        Koren Templeton
      </p>
      <p>
        
      </p>
      <p>
        On Stage
      </p>
    </div>
    <div
      className="participant"
    >
      <img
        alt="Avatar Here but no Avatar, sucks"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <p>
        Caty Flucker
      </p>
      <p>
        
      </p>
      <p>
        In Session
      </p>
    </div>
    <div
      className="participant"
    >
      <img
        alt="Avatar Here but no Avatar, sucks"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <p>
        Frank Runciman
      </p>
      <p>
        
      </p>
      <p>
        In Session
      </p>
    </div>
    <div
      className="participant"
    >
      <img
        alt="Avatar Here but no Avatar, sucks"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <p>
        Ashla Attwool
      </p>
      <p>
        
      </p>
      <p>
        On Stage
      </p>
    </div>
  </div>
  <hr />
  <div />
  <hr />
  <div
    className="stage"
  >
    <div
      className="participant"
    >
      <img
        alt="Avatar Here but no Avatar, sucks"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <p>
        Koren Templeton
      </p>
      <p>
        
      </p>
      <p>
        On Stage
      </p>
    </div>
    <div
      className="participant"
    >
      <img
        alt="Avatar Here but no Avatar, sucks"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <p>
        Ashla Attwool
      </p>
      <p>
        
      </p>
      <p>
        On Stage
      </p>
    </div>
  </div>
</div>
`;
